### PR TITLE
Introduce achievement action constants

### PR DESCRIPTION
**Pull request by luchaos**
- Do not require/send user in ajax post payloads
- Move scripts from all.js to where they are used
- Minor cleanup

### CODE REVIEW

File: public/controlpanel.php [hunk 1]
```diff
@@ -395,22 +395,18 @@ function validateEmail() {
...
+            if ($errorCode == 'badnewpass') {
+                echo "<div id=\"warning\">Info: Errors changing your password, passwords too short!</div>";
+            }
+            if ($errorCode == 'passinequal') {
+                echo "<div id=\"warning\">Info: Errors changing your password, new passwords were not identical!</div>";
+            }
+            if ($errorCode == 'badpassold') {
+                echo "<div id=\"warning\">Info: Errors changing your password, old password was incorrect!</div>";
+            }
+            if ($errorCode == 'changepassok') {
```
Comment (Jamiras) [inline public/controlpanel.php:408]: these should probably be `elseif`s - only one will ever match at a time.

File: src/AchievementAction.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?php
+
+namespace RA;
+
+abstract class AchievementAction
```
Comment (Jamiras) [inline src/AchievementAction.php:5]: I think this should be `AchievementUpdateAction` or `Update` should be added to the individual actions. `AchievementAction::DisplayOrder` just doesn't convey the fact that DisplayOrder is being updated.

File: public/controlpanel.php (hunk 1 above)
Comment (luchaos) [reply public/controlpanel.php:408]: while that's true I personally dislike `if/elseif/else` chains. i favor clean and easy to ready constructs over minor performance gains.
i think this should be rewritten to a `match` clause where the message is set, and if, display a html element containing the message.
my goal in this case was to untangle the unnecessary indentations for future refactoring.

File: src/AchievementAction.php (hunk 2 above)
Comment (luchaos) [reply src/AchievementAction.php:5]: Fair point - this was one of the reasons for marking this PR as a draft eventually. The constants should include the verb similar to the other enums, i agree.
Yet, while working on this I found it odd to even specify the kind of update as an enum - these actions may as well be separate request files doing just one kind of update. That was the goal when moving request files in the first place. I'll probably refactor all of those update requests to get rid of the "action" constants where possible and make their usage in scripts obsolete by calling dedicated endpoints explicitly.

**Comment by luchaos** [comment]
Outdated and superceded by upcaoming v2 updates. Closing.

**Pull request closed without merging**